cudamapper/include/claragenomics/cudamapper/overlapper.hpp

```diff
@@ -67,9 +67,11 @@ class Overlapper
     /// \param target_parser Parser for target reads
     /// \param num_alignment_engines Number of parallel alignment engines to use for alignment
     /// \param cigar Output vector to store CIGAR string for alignments
+    /// \param allocator The allocator to allocate memory on the device
     static void align_overlaps(std::vector<Overlap>& overlaps, const claragenomics::io::FastaParser& query_parser,
                                const claragenomics::io::FastaParser& target_parser, int32_t num_alignment_engines,
-                               std::vector<std::string>& cigar);
+                               std::vector<std::string>& cigar,
+                               DefaultDeviceAllocator allocator);
```

Comment:
In cudamapper we put allocator as first parameter, so I would move it for consistency.
Whether that is the right place for it is another question.

Reply:
Yes, agreed.
On a related note: I think we should move this function out of `overlapper.hpp` and put it in `main.cu`. The reason is that this function introduces unnecessary dependency of the cudamapper library on libcudaaligner. If we move it to main.cu, only the executable `cudamapper` will depend on `libcudaaligner`. Opinions @tijyojwad @mimaric @vellamike ?

Reply:
Yep sounds good to me

Reply:
done.